### ISSUE TITLE
feat: class_urls could be used to point to the class definitions

### ISSUE THREAD

**Issue by markdoerr** [nfdi4cat/pid4cat-model#19]
during transformation of the model to other targets, class_urls might help to point back to the original definition and could be added.

**Comment by dalito** [comment]
LinkML model docs for [class_uri](https://linkml.io/linkml-model/latest/docs/class_uri/): *URI of the class that provides a semantic interpretation of the element in a linked data context.*

@markdoerr @HendrikBorgelt  To which ontologies and vocabularies should we preferably link in [class_uri](https://linkml.io/linkml-model/latest/docs/class_uri/), [slot_uri](https://linkml.io/linkml-model/latest/docs/slot_uri/), [meaning](https://linkml.io/linkml-model/latest/docs/meaning/): FOAF, PROV, DCAT, spar/datacite, ..., voc4cat? (I am not a fan of schema.org for science.)

How important is to map the classes to exactly matching owl ontology class definitions? Should the mapping be suitable for reasoning? What to do if there is no suitable mapping?

These are more general questions. We can nevertheless start with gradually adding class_uri, slot_uri, meaning to the schema.

**Comment by HendrikBorgelt** [comment]
The simplest answer first, since the PID is not large enough of a data model to be valuable for its own logic (small pid4cat ontology), a mapping must be implemented to map from this the pid data schema into another terminology. As such it is helpful if we use Ontology classes, but depending on who we present these classes, people will not care for their precise definition and thus enter whatever suits them best.

| format | representation in a pid | usefullness for pid creators |
|---|---|---|
| plain name | class | limited, because you need to find the definition somewhere in the documentation |
| curie name | owl:class | people might suspect that something special is going on but the are not required to read through the definition. In my mind the best version |
| plain URI | http://www.w3.org/2002/07/owl#Class | people know that they must read whats going on, and will be enticed to align the respective defintion. However for "Programming/Semantic beginners" this is pure overkill |

regarding the question "To which ontologies and vocabularies should we preferably link...", I would yes, whatever is suited best. so the list you have mentioned sounds fine (FOAF, PROV, DCAT, spar/datacite, ..., voc4cat). 

Since I am not perfectly sure which entries must have a class_uri, slot_uri, meaning, I would appreciate it if you could shortly point me to the MD file where I could find them. I would then make a list of feasible terminology based on best matches in [LOV](https://lov.linkeddata.es/dataset/lov).

**Comment by dalito** [comment]
LinkML provides by default `class_uri` and `slot_uri` if they are not provided. From the [docs](https://linkml.io/linkml/schemas/uris-and-mappings.html#class-uri-and-slot-uri):

> If class and slot uris are omitted, then they are still generated behind the scenes, using the [default_prefix](https://w3id.org/linkml/default_prefix) slot at the schema level.

I believe this is fine for `pid4cat-model`. Mapping pid4cat enums values is also tricky, for example "SAMPLE" is a classifying category for a pid4cat PID type. This is not matching with the meaning of [sample in SOSA](http://www.w3.org/ns/sosa/Sample). The category is not something on which observations are made. The link to SOSA is better expressed as a [close_mapping](https://linkml.io/linkml-model/latest/docs/close_mappings/).

In summary I suggest to
- by default use the URIs for classes and slots that are automatically provided by LinkML.
- use SKOS-style [mappings](https://linkml.io/linkml-model/latest/docs/mappings/) to provide a mapping while avoiding to commit to completely reusing a linked data concept.
- use `meaning` for enums members if a linked data concept can be reused else use a mapping. Both mappings and meaning are optional.

It is interesting that the huge [biolink_model](https://github.com/biolink/biolink-model/blob/master/src/biolink_model/schema/biolink_model.yaml) does not use `class_uri` at all and has only few slots with `slot_uri`.
